src/components/Waveform.tsx

```diff
@@ -60,11 +60,11 @@ export default observer(function Waveform() {
 
   useEffect(() => {
     if (wavesurferRef.current) {
-      const progress =
-        timer.lastCursorPosition / wavesurferRef.current.getDuration();
+      const duration = wavesurferRef.current.getDuration();
+      const progress = duration > 0 ? timer.globalTime / wavesurferRef.current.getDuration() : 0;
       wavesurferRef.current.seekTo(progress);
     }
-  }, [timer.lastCursorPosition]);
+  }, [timer.globalTime]);
```

Comment:
Hmm, one side effect of this change is that this component will now be subscribed to changes in `timer.globalTime`. This is because `globalTime` is unobservable property (basically all of the properties on Store, uiStore, Timer, and Block are btw, due to `makeAutoObservable()`) and this component is an `observer()`. So if you console.log() inside this component, I think we would see it being rendered 60 times per second.

I've kind of made a mess of the new time variables (lastCursorPosition, globalTime, etc), but if we can basically accomplish the same thing here without using globalTime, that would be great. More of future proofing than anything though. Just want to stay ahead of performance problems which I caused in the past due to too many re renders.

Reply:
I see!

I just pushed a change to subscribe it to lastCursorPosition and _lastStartedAtDateTime instead (not sure if the _ is supposed to be "private" haha).

Now it seems to work, and I think it only will render if setTime is called, rather than on tick.

Reply:
Nice! I wonder if we can further trim away _lastStartedAtDateTime from the dependency array? I believe it is only changed when lastCursorPosition is changed, so maybe it's redundant. _lastStartedAtDateTime is just a weird thing that I came up with to solve a different problem and I don't think it needs to leak out of Timer. 

And yeah I imagined it as a private variable, meant to add private as well as the underscore. _globalTime as well maybe?

Reply:
Made that small change, otherwise looks good, merging